annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckFactory.java

```diff
@@ -189,7 +201,9 @@ public ConstraintCheckFactory(Types typeUtils, ConstraintHelper constraintHelper
 						new TargetCheck( annotationApiHelper ),
 						new ConstraintValidatorCheck( constraintHelper, annotationApiHelper ),
 						new AnnotationTypeMemberCheck( annotationApiHelper, typeUtils ),
-						new CrossParameterConstraintCheck( annotationApiHelper, constraintHelper, typeUtils )
+						new CrossParameterConstraintCheck( annotationApiHelper, constraintHelper, typeUtils ),
+						new AnnotationUserMessageCheck( annotationApiHelper ),
+						new AnnotationDefaultMessageCheck( annotationApiHelper, elementUtils )
```

Comment:
Yes ... you are correct here - I've just separated the check into two as you said, but didn't give much thought about it and just replaced the previous check with two new ones...
But `AnnotationDefaultMessageCheck` should only be present in `annotationTypeChecks` right ? 

I'll change the previous commit to have default check only in annotation checks.